src/historywork/VerifyBucketWork.cpp

```diff
@@ -16,37 +16,77 @@
 
 namespace stellar
 {
-
 VerifyBucketWork::VerifyBucketWork(
-    Application& app, WorkParent& parent,
+    Application& app, std::function<void()> callback,
     std::map<std::string, std::shared_ptr<Bucket>>& buckets,
     std::string const& bucketFile, uint256 const& hash)
-    : Work(app, parent, std::string("verify-bucket-hash ") + bucketFile,
-           RETRY_NEVER)
+    : Work(app, callback, "verify-bucket-hash-" + bucketFile, RETRY_NEVER)
     , mBuckets(buckets)
     , mBucketFile(bucketFile)
     , mHash(hash)
-    , mVerifyBucketSuccess{app.getMetrics().NewMeter(
-          {"history", "verify-bucket", "success"}, "event")}
-    , mVerifyBucketFailure{app.getMetrics().NewMeter(
-          {"history", "verify-bucket", "failure"}, "event")}
 {
-    fs::checkNoGzipSuffix(mBucketFile);
 }
 
 VerifyBucketWork::~VerifyBucketWork()
 {
-    clearChildren();
+}
+
+BasicWork::State
+VerifyBucketWork::doWork()
+{
+    // FIXME (mlo) this a logical problem, as this check shouldn't be here
+    // VerifyBucketWork shouldn't be required to know that it depends on
+    // GetAndUnzipRemoteFileWork. This will be resolved with batching though,
+    // as DownloadBuckets work will take care of when to dispatch download
+    // and verify tasks.
```

Comment:
Btw, this is an example of a more general consequence that we get with this Work design: if a work did not dispatch any children, if shouldn't care if it has any dependent children. In this example, `VerifyBucketWork` depends on `GetAndUnzipRemoteFileWork`, because higher level work `DownloadBucketsWork` created this dependency. I think as we move on from this, we should have any parent work (in this case `DownloadBucketsWork`, or maybe even a new class `DownloadAndVerifyBucketWork`) be in charge of adding and tracking status of its children.